Added Environment variables support

ishanjain28 opened the pull request:
Added Environment Variables support using `viper` and `pflag`. 

This fixes #84

Ullaakut commented:
Hey! I hope you don't mind, I made a few changes not necessarily related to this PR but that helped me test it better:
- Split the main file into 4 files (`cameradar.go`, `print.go`, `options.go` and `spinner.go`)
- Fixed the help message (calling `cameradar --help` printed `pflag: help requested`)
- Fixed targets by making it required again (not specifying it did launch nmap without a target and did not print an error message)
- Added more logs when logs enabled

The only problem I see remaining and that I don't really have time to work on at the moment is that I think that the environment should override the command line in the case of booleans and it's currently not the case, which means that if you run this code in a docker container and specify `CAMERADAR_LOGS=true` in your environment, it will not have any effect.

The command line is currently overriding the environment, and an unspecified boolean flag means it's set to false :/

If you see an easy way to fix this, it would be good to do it for the logs, as this is a major issue IMO for the release of a deployment-ready cameradar image. Otherwise, the quick & dirty way to fix this is to add a simple os.GetEnv() on the `CAMERADAR_LOGS` variable to see if it's set to true.

ishanjain28 commented:
Hi @Ullaakut 
I was reading `viper` docs again, And I had implemented `SetEnvPrefix` incorrectly. It doesn't works when I am also providing names of Environment Variables. So, I fixed that in this commit and implemented what you had suggested. 

Where should I add information about environment variables in Readme? Should I create a section right after `Command line options` and add it there?

Ullaakut commented:
Yes, seems like a good place to put them! :-) For the format to use for this section, you can follow what is done on the READMEs of Docker images on Docker Hub such as https://hub.docker.com/_/mysql/

I will check your latest commit, thanks!

ishanjain28 commented:
ping @Ullaakut

Ullaakut commented:
I did a few changes to the README, fixed the conflicts and squashed the commits, tell me if it's OK with you and I'll merge!

Thanks again for the contribution! :-)

ishanjain28 commented:
@Ullaakut It's okay.
